### PR TITLE
stats: fix handling of nan values in T tests and kendalltau

### LINKED ISSUE #6134: T test does not handle nans
The bug is in line 3885, in file scipy/scipy/stats/stats.py :

```
[contains_nan, nan_policy = (_contains_nan(a, nan_policy) or
                            _contains_nan(b, nan_policy))
```
must be

```
contains_nan             = _contains_nan(a, nan_policy)[0] or
                            _contains_nan(b, nan_policy)[0]
```

since logic apply good only on Boolean values.

See [http://stackoverflow.com/questions/37022888/t-test-in-scipy-with-nan-values/37026951](url)
for an exemple.

### PR DESCRIPTION
**Pull request by ev-br**
closes https://github.com/scipy/scipy/issues/6134

<s>Still need to address `ttest_rel` and possibly `kendalltau`.</s> done.

### CODE REVIEW

**Comment by ev-br** [comment]
I'm going to merge this by myself, caveman-style.

File: scipy/stats/stats.py [hunk 1]
```diff
@@ -3663,13 +3670,19 @@ def ttest_rel(a, b, axis=0, nan_policy='propagate'):
...
+        nan_policy = 'omit'
 
     if contains_nan and nan_policy == 'omit':
         a = ma.masked_invalid(a)
-        return mstats_basic.ttest_rel(a, b, axis)
+        b = ma.masked_invalid(b)
+        m = ma.mask_or(ma.getmask(a), ma.getmask(b))
+        aa = ma.array(a, mask=m, copy=True)
+        bb = ma.array(b, mask=m, copy=True)
+        return mstats_basic.ttest_rel(aa, bb, axis)
```
Comment (josef-pkt) [inline scipy/stats/stats.py]: this will return a masked array, not a numpy array with nans ?

**Comment by josef-pkt** [comment]
@ev-br  Sorry, I have problems keeping my priorities straight

Looks good to me, 
except that the `stats.stats` function returns a masked array, AFAICS
although that wasn't changed by this PR

File: scipy/stats/stats.py (hunk 1 above)
Comment (ev-br) [inline scipy/stats/stats.py]: Yup. Which is consistent with what a bunch of other functions do for `nan_policy='omit'`. Which is likely an oversight (Abraham and Ralf did not get much help last summer when this all was being added; I certainly did not help much).

We can consider this a bug and blanket convert these masked arrays to ndarrays-with-nans, across all `stats.py`.  scipy 0.18 would be a nice target for this IMO.

**Pull request merged**